service/collector/endpoint.go

```diff
@@ -47,23 +61,215 @@ func NewEndpoint(config EndpointConfig) (*Endpoint, error) {
 		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
 	}
 
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		MaxIdleConns:    maxIdleConnection,
+	}
+
+	client := &http.Client{
+		Transport: tr,
+		Timeout:   maxTimeoutSec * time.Second,
+	}
+
 	i := &Endpoint{
-		g8sClient: config.G8sClient,
-		k8sClient: config.K8sClient,
-		logger:    config.Logger,
+		g8sClient:     config.G8sClient,
+		httpClient:    client,
+		httpTransport: tr,
+		k8sClient:     config.K8sClient,
+		localIP:       getLocalIP(),
+		logger:        config.Logger,
 	}
 
 	return i, nil
 }
 
-func (i *Endpoint) Collect(ch chan<- prometheus.Metric) error {
+func (e *Endpoint) Collect(ch chan<- prometheus.Metric) error {
 	// TODO
 	// Here we will put implementation of collecting data for each endpoint IP
+	listOpts := metav1.ListOptions{}
+	getOpts := metav1.GetOptions{}
+
+	kvmConfigs, err := e.g8sClient.ProviderV1alpha1().KVMConfigs(crdNamespace).List(listOpts)
```

Comment:
You defined RBAC rules for all provider CRs. Now the collector implementation is type specific. I think it would be good to have the abstract lister interface injected into the collector and a type assertion. You can see an example of this in https://github.com/giantswarm/statusresource/blob/master/status_collector.go. The difference there is that the watcher interface is configured, and not the lister. We want to move to lister interfaces though.

Reply:
The endpoint-collector make sense only for KVM, other collectors maybe be created for other providers, so i think having the only KVM there make sense.

Reply:
I can remove the RBAC rules fro other provider for now as its not necessery